spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,39 @@
+require "rails_helper"
+
+RSpec.describe "the /recipes/:id show page" do
+  describe "When I visit /recipes/:id" do
+    before :each do
+      @recipe_1 = Recipe.create!(name: "Lasagna", complexity: 2, genre: "Italian")
+
+      @ingredient_1 = Ingredient.create!(name: "Pasta", cost: 2)
+      @ingredient_2 = Ingredient.create!(name: "Cheese", cost: 4)
+      @ingredient_3 = Ingredient.create!(name: "Sauce", cost: 3)
+
+      RecipeIngredient.create!(recipe: @recipe_1, ingredient: @ingredient_1)
+      RecipeIngredient.create!(recipe: @recipe_1, ingredient: @ingredient_2)
+      RecipeIngredient.create!(recipe: @recipe_1, ingredient: @ingredient_3)
+    end
+
+    # User Story 2
+    it "shows the name, complexity and genre of recipe" do
+      visit "/recipes/#{@recipe_1.id}"
+
+      expect(page).to have
+    end
+
+    # User Story 2
+    xit "displays recipe's: name, complexity, genre and ingredient list" do
```

Comment:
This looks very close. I would split the tests. One test for name, complexity, genre. One test for list of ingredients. 
```ruby
expect(page).to have_content("Name: #{recipe_1.name}")
expect(page).to have_content("Complexity: #{recipe_1.complexity}")
expect(page).to have_content("Genre: #{recipe_1.genre}")
```
Then
```ruby
within "#recipe_ingredients" do 
  expect(page).to have_content("Pasta")
  expect(page).to have_content("Cheese")
  expect(page).to have_content("Sauce")
end
```